app/controllers/users_controller.rb

```diff
@@ -26,4 +37,13 @@ def show
   def user_params
     params.require(:user).permit!
   end
+
+  def handle_invitation
+    if params[:invitation_token]
+      invitation = Invitation.find_by(token: params[:invitation_token])
+      invitation.inviter.follow(@user)
+      @user.follow(invitation.inviter)
+      invitation.update_column(:token, nil)
```

Comment:
Yep. Did the same thing when updating a user's password... was acting on updating the `user` directly in the `ResetPasswordsController`. Delegated it to the model.
One thing I could not do however was use the `Tokenable` module for user as I am not creating the token on user during creation. It is only when a password reset is requested. So the Tokenable concern is only used for Invitation right now. Any work around to avoid duplication due to this?